cli/cmd/diagnose.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"github.com/odigos-io/odigos/cli/cmd/resources"
+	"github.com/odigos-io/odigos/cli/pkg/kube"
+	"github.com/spf13/cobra"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"path/filepath"
+)
+
+const (
+	mainDir       = "odigos-diagnose"
+	logDir        = "logs"
+	logBufferSize = 1024 * 1024 // 1MB buffer size for reading logs in chunks
+)
+
+var diagnozeCmd = &cobra.Command{
+	Use:   "diagnose",
+	Short: "Diagnose Client Cluster",
+	Long:  `Diagnose Client Cluster to identify issues and resolve them. This command is useful for troubleshooting and debugging.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		client, err := kube.CreateClient(cmd)
+		if err != nil {
+			kube.PrintClientErrorAndExit(err)
+		}
+
+		err = startDiagnose(ctx, client)
+		if err != nil {
+			fmt.Printf("The diagnose script crashed on: %v\n", err)
+		}
+	},
+}
+
+func startDiagnose(ctx context.Context, client *kube.Client) error {
+	if err := createAllDirs(); err != nil {
+		return err
+	}
+
+	if err := fetchOdigosComponentsLogs(ctx, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createAllDirs() error {
+	if err := os.RemoveAll(mainDir); err != nil {
```

Comment:
Also related to Eden's comment, you can have a look at [the TempDir function](https://pkg.go.dev/os#TempDir).
I think having the files created in a temp location and only saving the final zip in the CWD is the accepted behavior

Reply:
TempDir is deprecated from go 1.16 - the new way is with os.MkdirTemp

Reply:
ohhh my bad, ioutil.TempDir() is deprecated